### PR TITLE
test: Condition tests for serialize

### PR DESCRIPTION
**Pull request by seebees**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: modules/serialize/src/aad_factory.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export function aadFactory (fromUtf8: (input: string) => Uint8Array) {
           ? ContentAADString.FINAL_FRAME_STRING_ID
           : ContentAADString.FRAME_STRING_ID
       default:
-        throw new Error('bad')
+        throw new Error('Unrecognized context type')
```
Comment (mattsb42-aws) [inline modules/serialize/src/aad_factory.ts]:
```suggestion
        throw new Error('Unrecognized content type')
```

Comment (mattsb42-aws) [reply modules/serialize/src/aad_factory.ts]: Also, should this be a particular type of error?

File: modules/serialize/src/decode_body_header.ts [hunk 2]
```diff
@@ -148,7 +148,8 @@ export function decodeFinalFrameBodyHeader (buffer: Uint8Array, headerInfo: Head
 
   /* The precondition SEQUENCE_NUMBER_END: Uint32(FFFF) is handled above. */
   const sequenceEnd = dataView.getUint32(readPos, false) // big endian
-  needs(sequenceEnd === SequenceIdentifier.SEQUENCE_NUMBER_END, '')
+  /* Postcondition: sequenceEnd must be SEQUENCE_NUMBER_END. */
+  needs(sequenceEnd === SequenceIdentifier.SEQUENCE_NUMBER_END, 'Malformed sequenceEnd')
```
Comment (mattsb42-aws) [inline modules/serialize/src/decode_body_header.ts]: This seems too specific and unlikely that an arbitrary developer is going to understand what it means.
```suggestion
  needs(sequenceEnd === SequenceIdentifier.SEQUENCE_NUMBER_END, 'Malformed final frame: Invalid sequence number end value')
```

File: modules/serialize/src/aad_factory.ts (hunk 1 above)
Comment (seebees) [reply modules/serialize/src/aad_factory.ts]: I have not gotten to "types of errors" _yet_.  Error codes are a thing :(

**Pull request merged**